core/config/config.go

```diff
@@ -20,201 +19,358 @@ import (
+	SlasherAddr           common.Address
 }
 
-// These are read from AlignedLayerDeploymentFileFlag
-type AlignedLayerDeploymentRaw struct {
-	Addresses AlignedLayerContractsRaw `json:"addresses"`
+type EigenLayerDeploymentConfigFromJson struct {
+	Addresses struct {
+		DelegationManagerAddr common.Address `json:"delegationManager"`
+		AVSDirectoryAddr      common.Address `json:"avsDirectory"`
+		SlasherAddr           common.Address `json:"slasher"`
+	} `json:"addresses"`
 }
 
-type AlignedLayerContractsRaw struct {
-	AlignedLayerServiceManagerAddr         string `json:"alignedLayerServiceManager"`
-	AlignedLayerRegistryCoordinatorAddr    string `json:"registryCoordinator"`
-	AlignedLayerOperatorStateRetrieverAddr string `json:"operatorStateRetriever"`
+type AggregatorConfig struct {
+	BaseConfig *BaseConfig
+	Aggregator struct {
+		AggregatorServerIpPortAddress string         `yaml:"aggregator_server_ip_port_address"`
+		BlsPublicKeyCompendiumAddress common.Address `yaml:"bls_public_key_compendium_address"`
+		AvsServiceManagerAddress      common.Address `yaml:"avs_service_manager_address"`
+		EnableMetrics                 bool           `yaml:"enable_metrics"`
+	} `yaml:"aggregator"`
 }
 
-// NewConfig parses config file to read from flags or environment variables
-// Note: This config is shared by challenger and aggregator, so we put in the core.
-// Operator has a different config and is meant to be used by the operator CLI.
-func NewConfig(ctx *cli.Context) (*Config, error) {
-	var configRaw ConfigRaw
+type OperatorConfig struct {
+	BaseConfig *BaseConfig
+	Operator   struct {
+		Address                   common.Address `yaml:"address"`
+		EarningsReceiverAddress   common.Address `yaml:"earnings_receiver_address"`
+		DelegationApproverAddress common.Address `yaml:"delegation_approver_address"`
+		StakerOptOutWindowBlocks  int            `yaml:"staker_opt_out_window_blocks"`
+		MetadataUrl               string         `yaml:"metadata_url"`
+		RegisterOperatorOnStartup bool           `yaml:"register_operator_on_startup"`
+	} `yaml:"operator"`
+}
 
-	configFilePath := ctx.GlobalString(ConfigFileFlag.Name)
-	if configFilePath != "" {
-		err := sdkutils.ReadYamlConfig(configFilePath, &configRaw)
-		if err != nil {
-			fmt.Println("Could not read yaml config file")
-			return nil, err
-		}
+var (
+	// Required Flags
+	BaseConfigFileFlag = cli.StringFlag{
+		Name:     "base-config-file",
+		Required: true,
+		Usage:    "Load base configurations from `FILE`",
 	}
+	AggregatorConfigFileFlag = cli.StringFlag{
+		Name:     "aggregator-config-file",
+		Required: true,
+		Usage:    "Load aggregator configurations from `FILE`",
+	}
+	OperatorConfigFileFlag = cli.StringFlag{
+		Name:     "operator-config-file",
+		Required: true,
+		Usage:    "Load operator configurations from `FILE`",
+	}
+	// Optional Flags
+)
+
+func NewAggregatorConfig(baseConfigFilePath, aggregatorConfigFilePath string) (*AggregatorConfig, error) {
```

Comment:
If we are using log.Fatal instead of returning error, remove error from return type. Two options here:

1. Go back to return err, and log error instead of fatal (which kills process)
2. Return just *AggregatorConfig